avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1445,7 +1446,7 @@ private Schema toConnectSchema(org.apache.avro.Schema schema, boolean forceOptio
       while (paramIt.hasNext()) {
         Map.Entry<String, JsonNode> field = paramIt.next();
         JsonNode jsonValue = field.getValue();
-        if (!jsonValue.isTextual()) {
+        if ((!field.getKey().endsWith(DOC_PROP) && !jsonValue.isTextual())) {
```

Comment:
@robin7m Right, I get that this is the case you found, but I am wondering if we really want checks specific to docs when it seems there's a more general solution and the original intent was that we wouldn't try to store `null`s in the first place.